layers/gpuav/spirv/sanitizer_pass.cpp

```diff
@@ -0,0 +1,130 @@
+/* Copyright (c) 2024-2025 LunarG, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "sanitizer_pass.h"
+#include "containers/container_utils.h"
+#include "module.h"
+#include <spirv/unified1/spirv.hpp>
+#include <iostream>
+
+#include "generated/gpuav_offline_spirv.h"
+
+namespace gpuav {
+namespace spirv {
+
+const static OfflineModule kOfflineModule = {instrumentation_sanitizer_comp, instrumentation_sanitizer_comp_size,
+                                             UseErrorPayloadVariable};
+
+const static OfflineFunction kOfflineFunction = {"inst_sanitizer_divide_by_zero", instrumentation_sanitizer_comp_function_0_offset};
+
+SanitizerPass::SanitizerPass(Module& module) : Pass(module, kOfflineModule) {}
+
+// By appending the LinkInfo, it will attempt at linking stage to add the function.
+uint32_t SanitizerPass::GetLinkFunctionId() { return GetLinkFunction(link_function_id_, kOfflineFunction); }
+
+uint32_t SanitizerPass::CreateFunctionCall(BasicBlock& block, InstructionIt* inst_it, const InstructionMeta& meta) {
+    const uint32_t function_result = module_.TakeNextId();
+    const uint32_t function_def = GetLinkFunctionId();
+    const uint32_t bool_type = module_.type_manager_.GetTypeBool().Id();
+
+    const uint32_t inst_position = meta.target_instruction->GetPositionOffset();
+    const uint32_t inst_position_id = module_.type_manager_.CreateConstantUInt32(inst_position).Id();
+
+    uint32_t divisor_id = meta.target_instruction->Word(4);
+    if (meta.target_instruction->Opcode() == spv::OpSDiv) {
+        const Type& uint32_type = module_.type_manager_.GetTypeInt(32, false);
+        const uint32_t new_id = module_.TakeNextId();
+        block.CreateInstruction(spv::OpBitcast, {uint32_type.Id(), new_id, divisor_id}, inst_it);
+        divisor_id = new_id;
+    }
+
+    block.CreateInstruction(spv::OpFunctionCall, {bool_type, function_result, function_def, inst_position_id, divisor_id}, inst_it);
+    module_.need_log_error_ = true;
+    return function_result;
+}
+
+bool SanitizerPass::RequiresInstrumentation(const Function& function, const Instruction& inst, InstructionMeta& meta) {
```

Comment:
Since it is not used
```suggestion
bool SanitizerPass::RequiresInstrumentation(const Instruction& inst, InstructionMeta& meta) {
```